common/url.go

```diff
@@ -615,8 +609,29 @@ func (c *URL) Clone() *URL {
 	return newUrl
 }
 
+func (c *URL) CloneWithParams(reserveParams []string, methods []string) *URL {
```

Comment:
Should add a  comment for  exported method

Reply:
Add comment.